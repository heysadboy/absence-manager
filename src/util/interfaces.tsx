export interface IAbsence {
    admitterId: number | null;
    admitterNote: string | null,
    confirmedAt: string | null,
    createdAt: string | null,
    crewId: number,
    endDate: string | null,
    id: number,
    memberNote: string | null,
    rejectedAt: string | null,
    startDate: string | null,
    type: string | null,
    userId: number
}

export interface IMember {
    crewId: number,
    id: number,
    image: string,
    name: string,
    userId: number
}

export interface IMemberList {
    [id: number]: IMember
}