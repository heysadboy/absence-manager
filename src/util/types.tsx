export type TPageIndex = {
    start: number;
    end: number;
    total: number
}

export type TDateRange = {
    start: string,
    end: string
}