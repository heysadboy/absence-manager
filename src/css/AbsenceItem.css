#member-image {
    width: 50px;
    height: 50px;
    margin: 10px;
}

#member-name {
    font-size: medium;
    font-weight: bolder;
}

#absence-type {
    text-transform: capitalize;
    margin-right: 10px;
}

@media screen and (max-width: 480px) {
    #member-image {
        width: 30px;
        height: 30px;
        margin: 5px;
    }
}