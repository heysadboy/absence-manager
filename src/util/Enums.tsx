export enum EAbsenceStatus {
    requested = 'Requested',
    confirmed = 'Confirmed',
    rejected = 'Rejected'
}

export enum EPageStatus {
    loading = 'Loading',
    error = 'Error',
    success = 'Success',
    empty = 'Empty'
}

export enum EAbsenceType {
    sickness = 'Sickness',
    vacation = 'Vacation',
    all = 'All'
}
