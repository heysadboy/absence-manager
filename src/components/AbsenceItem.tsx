import { FC } from 'react';
import { IMember, IAbsence } from '../util/Interfaces';
import AbsenceStatus from './AbsenceStatus';
import '../css/AbsenceItem.css';

interface IAbsenceItemProps {
    member: IMember;
    absence: IAbsence;
}

const AbsenceItem: FC<IAbsenceItemProps> = ({ member, absence }) => {
    return (
        <div id="absence-item" className="ui grid">
            <div className="three wide column">
                <img id="member-image" className="ui avatar image" alt={member.name} src={member.image} />
            </div>
            <div className="thirteen wide column">
                <div className="content">
                    <AbsenceStatus absence={absence} />
                    <span id="member-name">{member.name}</span>
                    <div id="member-description" className="description">
                        <i className="home icon" /> <span id="absence-type">{absence.type}</span><br />
                        <i className="calendar alternate outline icon" /> {absence.startDate} ~ {absence.endDate}<br />
                        {absence.memberNote !== null && absence.memberNote !== "" ? `Member Note: ${absence.memberNote}` : null}<br />
                        {absence.admitterNote !== null && absence.admitterNote !== "" ? `Admitter Note: ${absence.admitterNote}` : null}
                    </div>
                </div>
            </div>
        </div>
    )
};

export default AbsenceItem;