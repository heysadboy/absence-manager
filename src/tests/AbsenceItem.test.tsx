import renderer from 'react-test-renderer';
import  AbsenceItem from '../components/AbsenceItem';
import { IAbsence, IMember } from '../util/Interfaces';


test('AbsenceItem.tsx snapshot testing', () => {

  const member: IMember = {
    crewId: 1,
    id: 1,
    image: "image link",
    name: "name",
    userId: 1
  };

  const absence: IAbsence = {
    admitterId: 1,
    admitterNote: "admitter note",
    confirmedAt: "2020-12-12",
    createdAt: "2020-12-12",
    crewId: 1,
    endDate: "2020-12-12",
    id: 1,
    memberNote: "member note",
    rejectedAt: "2020-12-12",
    startDate: "2020-12-12",
    type: "vacation",
    userId: 1
  }

  const component = renderer.create(
    <AbsenceItem member={member} absence={absence}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});