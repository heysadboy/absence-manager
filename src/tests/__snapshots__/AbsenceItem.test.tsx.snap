// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbsenceItem.tsx snapshot testing 1`] = `
<div
  className="ui grid"
  id="absence-item"
>
  <div
    className="three wide column"
  >
    <img
      alt="name"
      className="ui avatar image"
      id="member-image"
      src="image link"
    />
  </div>
  <div
    className="thirteen wide column"
  >
    <div
      className="content"
    >
      <div
        className="ui mini green label"
      >
        Confirmed
      </div>
      <span
        id="member-name"
      >
        name
      </span>
      <div
        className="description"
        id="member-description"
      >
        <i
          className="home icon"
        />
         
        <span
          id="absence-type"
        >
          vacation
        </span>
        <br />
        <i
          className="calendar alternate outline icon"
        />
         
        2020-12-12
         - 
        2020-12-12
        <br />
        Member Note: member note
        <br />
        Admitter Note: admitter note
      </div>
    </div>
  </div>
</div>
`;
